packages/Adapters/POD_C/src/DTK_POD_PointCloudEntityImpl.cpp

```diff
@@ -0,0 +1,178 @@
+*/
+//---------------------------------------------------------------------------//
+/*!
+ * \brief DTK_STKMeshEntityImpl.cpp
+ * \author Stuart R. Slattery
+ * \brief Point cloud entity implementation.
+ */
+//---------------------------------------------------------------------------//
+
+#include <limits>
+
+#include "DTK_POD_PointCloudEntityImpl.hpp"
+#include "DTK_DBC.hpp"
+
+namespace DataTransferKit
+{
+//---------------------------------------------------------------------------//
+// Constructor.
+POD_PointCloudEntityImpl::POD_PointCloudEntityImpl(
+    const double* cloud_coords,
+    const unsigned num_points,
+    const int space_dim,
+    const DTK_Data_layout layout,
+    const EntityId global_id,                          
+    const int local_id,
+    const int owner_rank )
+    : d_cloud_coords( cloud_coords )
+    , d_offsets( space_dim, -1 )
+    , d_global_id( global_id )
+    , d_owner_rank( owner_rank )
+{
+    DTK_REQUIRE( DTK_INTERLEAVED == layout ||
+                 DTK_BLOCKED == layout );
+    
+    // Calculate the offsets into the coordinates array.
+    for ( int d = 0; d < space_dim; ++d )
+    {
+        d_offsets[d] = ( DTK_INTERLEAVED == layout )
+                       ? space_dim*local_id + d
+                       : d*num_points + local_id;
+    }
+}
+
+//---------------------------------------------------------------------------//
+// Get the coordinates of the point in a given dimension.
+double POD_PointCloudEntityImpl::coords( const int dim ) const
+{
+    DTK_REQUIRE( dim < physicalDimension() );
+    return d_coords[ d_offsets[dim] ];
+}
+
+//---------------------------------------------------------------------------//
+// Get the unique global identifier for the entity.
+EntityId POD_PointCloudEntityImpl::id() const
+{ 
+    return d_global_id;
+}
+    
+//---------------------------------------------------------------------------//
+// Get the parallel rank that owns the entity.
+int POD_PointCloudEntityImpl::ownerRank() const
+{ 
+    return d_owner_rank;
+}
+
+//---------------------------------------------------------------------------//
+// Get the topological dimension of the entity.
+int POD_PointCloudEntityImpl::topologicalDimension() const
+{
+    return 0;
+}
+
+//---------------------------------------------------------------------------//
+// Return the physical dimension of the entity.
+int POD_PointCloudEntityImpl::physicalDimension() const
+{ 
+    return d_offsets.size();
+}
+
+//---------------------------------------------------------------------------//
+// Return the Cartesian bounding box around an entity.
+void POD_PointCloudEntityImpl::boundingBox( Teuchos::Tuple<double,6>& bounds ) const
+{
+    for ( int d = 0; d < physicalDimension(); ++d )
+    {
+        bounds[d] = d_cloud_coords[ d_offsets[d] ];
+        bounds[d+3] = bounds[d];
+    }
+
+    for ( int d = physicalDimension(); d < 3; ++d )
+    {
+        bounds[d] = 0.0;
+        bounds[d+3] = bounds[d];
+    }
```

Comment:
This is so we get a default of 0 for the dimensions that don't exist. The bounding box is always 3 dimensions.